borsh-rs/borsh/src/ser/mod.rs

```diff
@@ -222,6 +244,12 @@ impl BorshSerialize for Box<[u8]> {
     }
 }
```

Comment:
We have boxed slice here, would probably make sense implement for `Box<str>` instead of `&str`.

Also, please add deserialize methods and some sanity tests.

Reply:
> please add deserialize methods 

Deserializing borrowed types requires more structural changes that could impact code size. related: #45